CMakeLists.txt

```diff
@@ -19,10 +19,10 @@ set(${PROJECT_NAME}_VERSION ${${PROJECT_NAME}_VERSION_MAJOR}.${${PROJECT_NAME}_V
 
 if(CMAKE_VERSION VERSION_LESS "3.1")
     if(CMAKE_COMPILER_IS_GNUCXX)
-        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
```

Comment:
I think you should remove this IF entirely since the minimum version you set to 3.12 already

Reply:
CMake's bump was related to a CMake warning, so I should probably remove this.  Now, on the topic of older versions of C++, since I didn't end up making use of `std::as_const` as I originally intended, we could continue to leave the minimum at C++11...

@dantti If I were to leave the minimum at 11, would we need to use a lower CMake version as well?

Reply:
I don't think so 3.12 is old enough to raise it I would say, and we are likely the only users of this library...

Reply:
I agree, 3.12 is old enough. I don't see a need for supporting C++ 11 but since we also don't need to bump it, we can leave it as it is until there's a need for a more modern version.